### ISSUE TITLE
(Flaky-Test) Possible race condition - Fix Prometheus Tests

### ISSUE THREAD

**Issue by devinbost** [apache/pulsar#6256]
These are intermittent errors that appear to be race conditions or concurrency issues, rather than issues with just the tests:

```
[ERROR] Tests run: 6, Failures: 2, Errors: 0, Skipped: 2, Time elapsed: 1.904 s <<< FAILURE! - in org.apache.pulsar.broker.stats.PrometheusMetricsTest
576[ERROR] testPerNamespaceStats(org.apache.pulsar.broker.stats.PrometheusMetricsTest) Time elapsed: 0.062 s <<< FAILURE!
577java.lang.AssertionError: expected [true] but found [false]
578 at org.testng.Assert.fail(Assert.java:96)
579 at org.testng.Assert.failNotEquals(Assert.java:776)
580 at org.testng.Assert.assertTrue(Assert.java:44)
581 at org.testng.Assert.assertTrue(Assert.java:54)
582 at


[ERROR] testPerTopicStats(org.apache.pulsar.broker.stats.PrometheusMetricsTest) Time elapsed: 0.071 s <<< FAILURE!
600java.lang.AssertionError: expected [true] but found [false]
601 at org.testng.Assert.fail(Assert.java:96)
602 at org.testng.Assert.failNotEquals(Assert.java:776)
603 at org.testng.Assert.assertTrue(Assert.java:44)
604 at org.testng.Assert.assertTrue(Assert.java:54)
605 at
```

They may reflect intermittent issues that are happening in production.

They can be seen here: https://github.com/apache/pulsar/pull/6202/checks?check_run_id=431093396

I've attached the logs. 

[logs_14078.zip](https://github.com/apache/pulsar/files/4168962/logs_14078.zip)
[surefire-artifacts (2).zip](https://github.com/apache/pulsar/files/4195074/surefire-artifacts.2.zip)

**Comment by devinbost** [comment]
I found the issue. There's a race in how we're getting metrics from Prometheus (because we're expecting a particular order, but Prometheus doesn't guarantee that order). There also might be a race between our tests, so I've broken them up into separate topics and filtered the results. The tests are passing locally.

**Comment by devinbost** [comment]
This is fixed in #6202 
I'm planning on splitting that big PR into smaller ones.

**Comment by tisonkun** [comment]
Closed as stale and no recent report. Please open a new issue if it's still relevant in maintained versions.
